pilot/pkg/serviceregistry/kube/controller/ambient/authorization_test.go

```diff
@@ -981,6 +989,70 @@ func TestWaypointPolicyStatusCollection(t *testing.T) {
 				},
 			},
 		},
+		{
```

Comment:
We should probably add a test case for what should happen when an authz policy is bound to a non-waypoint gatewayclass to codify the behavior